### PR TITLE
HIP : keep FA head size 192 off the WMMA kernel

### PR DESCRIPTION
**Pull request by SimonTeixidor**
Branch `pr/hip-rebase-fixes` (`3429e79ad`), 1 commit on `halo/master` `69946438a`.

## Overview

Route only head sizes that have WMMA device code (`DKQ <= 128`, `DKQ == 256`) to `BEST_FATTN_KERNEL_MMA_F16`. Upstream `16378d93f` widened the gate to `Q->ne[0] <= 256`, but the merge kept the Strix device-code guard that builds WMMA only for 256 above 128, so head size 192 reached a `NO_DEVICE_CODE` kernel and trapped. 192 goes back to the tile kernel. Selection for every other head size is unchanged.

## Measurements

```
Device:     HP ZBook Ultra G1a 14 inch Mobile Workstation, AMD RYZEN AI MAX+ PRO 395 w/ Radeon 8060S (gfx1151)
Memory:     64 GB LPDDR5X unified (62 GiB visible to Linux; speed not read, no root)
Power:      AC, ACPI platform_profile=balanced, amdgpu power_dpm_force_performance_level=auto (GPU clocks not pinned)
BIOS:       UMA carve-out 512 MiB VRAM, GTT 60 GiB
Kernel:     7.1.13-200.fc44.x86_64, cmdline amdgpu.gttsize=61440 ttm.pages_limit=15728640 amd_iommu=off
Backend:    ROCm 10.0.0 + patched HIP runtime 7.16.26353-c4b77ac5cb (pwilkin rocm-systems, retained PM4), image localhost/strix-halo:stackv2-2d7825b37 (df09292858be); env HSA_OVERRIDE_GFX_VERSION=11.5.1 GGML_HIP_ENABLE_UNIFIED_MEMORY=1 DEBUG_HIP_GRAPH_PM4=1; HIP_LAUNCH_BLOCKING unset
Build:      localhost/strix-halo:builder (f1db10cdcd29), CMAKE_BUILD_TYPE=Release GGML_HIP=ON GPU_TARGETS=gfx1151 GGML_HIP_GRAPHS=ON GGML_HIP_MMQ_MFMA=ON GGML_HIP_NO_VMM=ON GGML_CUDA_FA=ON LLAMA_BUILD_TESTS/EXAMPLES/TOOLS=ON, -j 20
Baseline:   654803517 (halo/master), built and run in this session
Change:     a0911a3a2 + since-dropped commits (see below)
Model:      Qwen3.8-27B-UD-IQ4_XS.gguf (sha256 40fac4050e94...), head size 256
```

| | stock | this PR |
|---|---|---|
| `test-backend-ops test -b ROCm0 -o FLASH_ATTN_EXT` | rc 139 after 3705 OK | passed |

Stock aborts with `HSA_STATUS_ERROR_EXCEPTION` on an existing grid case, `hsk=192,hsv=128,nh=4,nr23=[8,1],kv=512,nb=3`. With the fix all 354 hsk=192 cases pass. No speed change is expected, and Qwen3.8-27B (head size 256) gave byte-identical logits and greedy decodes against stock.

## Requirements

- I have read and agree with the [contributing guidelines](CONTRIBUTING.md)
- This change is justified by measurements on Strix Halo (gfx1151)
- AI usage disclosure: AGENT-AUTHORED. Claude (Claude Code) wrote the commit and this description and ran the measurements; the owner reviews.
- What was NOT verified:
  - Measured on `654803517` in a build that also carried two since-dropped commits (a `prop.integrated` re-apply in `ggml-cuda.cu` and extra FA test cases); neither touches FA kernel selection. This branch, rebased onto `69946438a`, was not re-run.
  - No head size 192 model run end to end (none on disk).
  - Full `test-backend-ops` suite not run, only `-o FLASH_ATTN_EXT`.
  - Vulkan and non-HIP CUDA not built.

https://claude.ai/code/session_01MkuKwgPNgpAg7EbxKR2iWC

### CODE REVIEW

**Review by dzannotti — APPROVED** [review]
Validation: Release HIP build for gfx1151 passed. test-backend-ops FLASH_ATTN_EXT with head size 192 passed 34/34 cases on Radeon 8060S/gfx1151.

Scope: this is a narrowly-scoped gfx1151 fallback and does not change attention math. Approved.

comment generated by my clanker Codex
